common/util/src/main/java/org/thingsboard/common/util/DonAsynchron.java

```diff
@@ -65,4 +67,32 @@ public static <T> ListenableFuture<T> submit(Callable<T> task, Consumer<T> onSuc
         return future;
     }
 
+    public static <T> ListenableFuture<T> submitWithTimeout(
+            Callable<T> task,
+            long timeout, TimeUnit timeUnit,
+            Executor executor,
+            ScheduledExecutorService timeoutExecutor
+    ) {
+        var future = Futures.submit(task, executor);
+        Futures.withTimeout(future, timeout, timeUnit, timeoutExecutor);
```

Comment:
Completely wrong. You should return the result of Futures.withTimeout. Also, this method is completely redundant

Reply:
There is no difference what we'll return - Futures.withTimeout will cancel the future-delegate, so we will getting the exception either on 'future' or Futures.withTimeout